### PR TITLE
ADAPT-966 - Record generic component completion statements

### PR DESCRIPTION
**Pull request by barry-learningpool**
Implemented generic statements for components (below is an example of the statement sent for a text component)

{
	"verb": {
		"id": "http://adlnet.gov/expapi/verbs/experienced",
		"display": {
			"de-DE": "erlebte",
			"en-US": "experienced",
			"fr-FR": "a éprouvé",
			"es-ES": "experimentó"
		}
	},
	"actor": {
		"objectType": "Agent",
		"account": {
			"homePage": "http://www.example.com",
			"name": "1625378"
		}
	},
	"object": {
		"id": "http://www.example.com/LA1/001/intro/component/56d6e8de211b207b14611e1a",
		"objectType": "Activity",
		"definition": {
			"name": {
				"en": "TEXT"
			},
			"type": "http://adaptlearning.org/component/text"
		}
	},
	"context": {
		"contextActivities": {
			"parent": {
				"id": "http://www.example.com/LA1/001/intro/assessment/56c303be8a8c5efe4704ccae"
			}
		},
		"language": "en",
		"registration": "760e3480-ba55-4991-94b0-01820dbd23a2"
	}
}

Implemented a 'StatementModel' base class and modified component and assessment statement models to extend this

### CODE REVIEW

**Comment by ryangadams** [comment]
+1 Looks good. I've not tested this for myself though.

File: js/adapt-xapi.js [hunk 1]
```diff
@@ -78,6 +80,25 @@ define(function(require) {
       this.listenTo(Adapt, "xapi:stateLoaded", this.restoreState);
     },
 
+    onComponentComplete: function(component) {
```
Comment (ryangadams) [inline js/adapt-xapi.js]: The onComponentComplete and onAssessmentComplete functions are the same apart from the new ComponentStatemetnModel vs new AssessmentStatementModel - could we combine them and create a StatementModelFactory (see my Java experience slipping in!)

**Comment by ryan-lynch** [comment]
+1

**Comment by bell87** [comment]
+1

**Pull request merged**